### DISCUSSION TITLE
Filter pushdown with parquet behaves differently depending on criterias ordering in the WHERE clause

### DISCUSSION THREAD

**Discussion by ericemc3** [duckdb/duckdb#10168] [Q&A]
### What happens?

2 logical identical requests consume different bandwiths depending on criterias ordering in the WHERE clause

### To Reproduce

The Parquet file is sorted on `codeCommuneEtablissement `but not on `siret `
```
SELECT geometry 
FROM 'https://static.data.gouv.fr/resources/sirene-geolocalise-parquet/20240107-143656/sirene2024-geo.parquet'
WHERE siret = '80818361000035' and codeCommuneEtablissement = '31555' ;
```
consumes 40 Mo

```
SELECT geometry 
FROM 'https://static.data.gouv.fr/resources/sirene-geolocalise-parquet/20240107-143656/sirene2024-geo.parquet'
WHERE codeCommuneEtablissement = '31555' and siret = '80818361000035';
```
consumes 10 Mo

Is there a way to optimize this, analyzing metadata to guess with column could be best sorted than the other?

Meanwhile, i understand that:
- it is important to sort parquet files according main filtering criterias
- knowing that, it is important to write WHERE clauses in the proper order

### OS:

Win11

### DuckDB Version:

9.2

### DuckDB Client:

shell wasm

### Full Name:

eric mauviere

### Affiliation:

icem7

### Have you tried this on the latest `main` branch?

I have tested with a main build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by szarnyasg** [comment]
Hello @ericemc3, you can query parquet metadata using the [`parquet_metadata` table function](https://duckdb.org/docs/data/parquet/metadata). You can also query
```sql
SELECT name, type FROM parquet_schema('...');
```
To get the column types.

However, neither really gives any direct indications on what you should sort on.

**Comment by ericemc3** [comment]
Yes, i mean, could DuckDB engine make the best guess, because ordinary people won't even realize that the order within the WHERE is important ?
